pkg/cloudprovider/providers/openstack/openstack.go

```diff
@@ -444,6 +445,65 @@ func getAddressByName(client *gophercloud.ServiceClient, name types.NodeName) (s
 	return addrs[0].Address, nil
 }
 
+// getInterfacesByName returns the node interfaces of the specified instance.
+func getInterfacesByName(client *gophercloud.ServiceClient, name types.NodeName) ([]attachinterfaces.Interface, error) {
+	var interfaces []attachinterfaces.Interface
+
+	// get server
+	srv, err := getServerByName(client, name)
+	if err != nil {
+		if err == ErrNotFound {
+			return interfaces, cloudprovider.InstanceNotFound
```

Comment:
@anguslees my point is: it is really confusing that a caller relies on the error type(or value?) to do different things:

1. as an implementer, I have to know how caller will invoke my code.
2. if the caller really cares whether the server exists or not, it is better to be explicit as a return value, and no error should be returned.

for example, `getInterfacesByName` returns an empty list along with a NotFoundError, what does this mean, is it really an error, does an empty list already imply NotFound?

I am not talking about the code here specifically though, it is a general idea.